WirescaleCli/WireguardConfigurationWriter.cs

```diff
@@ -0,0 +1,47 @@
+﻿using System.Diagnostics;
+using System.Text;
+
+namespace WirescaleCli;
+
+public class WireguardConfigurationWriter
+{
+    private readonly string _configurationFilePath;
+
+    public WireguardConfigurationWriter(string configurationFilePath)
+    {
+        _configurationFilePath = configurationFilePath;
+    }
+
+    public void Write(WireguardKeyPair wireguardKeyPair, WirescalePeerRegistration peerRegistration)
+    {
+        StringBuilder configurationBuilder = new StringBuilder();
+
+        configurationBuilder.AppendLine("[Interface]");
+        configurationBuilder.AppendLine($"PrivateKey = {wireguardKeyPair.PrivateKey}");
+        configurationBuilder.AppendLine($"Address = {peerRegistration.AllowedIps.Single()}");
+
+        configurationBuilder.AppendLine();
+
+        configurationBuilder.AppendLine("[Peer]");
+        configurationBuilder.AppendLine($"PublicKey = {peerRegistration.ServerPublicKey}");
+        configurationBuilder.AppendLine($"AllowedIPs = {peerRegistration.ServerAllowedIps.Single()}");
+        configurationBuilder.AppendLine($"Endpoint = {peerRegistration.ServerEndpoint}");
+
+        File.WriteAllText(_configurationFilePath, configurationBuilder.ToString());
+
+        //Move wireguard logic/configuration elsewhere
+        var process = Process.Start(new ProcessStartInfo
```

Comment:
If I'm already enrolled in the network, how is this situation handled?

Reply:
We should probably detect that situation before running this command, or even earlier (e.g. before authorizing the user in the first place, for instance based on the presence of the wireguard configuration file, or the interface being registered with wireguard already).  Implementing a `wirescale logout` command that'll clean up/tear down the local wireguard configuration, deregister the peer registration with the Wirescale API (if possible), etc would be nice.

Currently this CLI just generate and register a new public key every time it's executed, and the API doesn't support destroying a peer... Which will of course also cause issues allocating new peer IPs on the Wireguard server eventually :) Would love to implement the rest, but think that's outside the scope of this challenge -- BUT returning a relevant error message is not, and I've implemented a small change to address this [as mentioned in this comment](https://github.com/runesoerensen/wirescale/pull/5#discussion_r1213870802)